xm-commons-function/src/main/java/com/icthh/xm/commons/domain/spec/IFunctionSpec.java

```diff
@@ -0,0 +1,11 @@
+package com.icthh.xm.commons.domain.spec;
+
+import com.icthh.xm.commons.domain.enums.FunctionTxTypes;
+
+public abstract class IFunctionSpec {
```

Comment:
Please clarify why the name starts with "I"? seems its more logical to start with "Abstract"

Reply:
`IFunctionSpec` is shorter and is associated with interface. This class is abstract, because you can't use it for generics, only -> `AbstractFunctionService<FS extends IFunctionSpec>`

IFunctionSpec defines methods for all implementations without any implementations. That's why de facto it's an interface, de jure it's abstract class